src/include/sof/audio/tdfb/tdfb_comp.h

```diff
@@ -43,16 +43,24 @@
 
 struct tdfb_comp_data {
 	struct fir_state_32x16 fir[SOF_TDFB_FIR_MAX_COUNT]; /**< FIR state */
+	//struct sof_ipc_comp_event event;
+	//struct ipc_msg *msg;
 	struct comp_data_blob_handler *model_handler;
 	struct sof_tdfb_config *config;	    /**< pointer to setup blob */
+	struct sof_tdfb_angle *filter_angles;
+	struct sof_tdfb_mic_location *mic_locations;
 	int32_t in[TDFB_IN_BUF_LENGTH];	    /**< input samples buffer */
 	int32_t out[TDFB_IN_BUF_LENGTH];    /**< output samples mix buffer */
+	uint32_t az_value;		    /**< beam steer azimuth as in control enum */
 	int32_t *fir_delay;		    /**< pointer to allocated RAM */
 	int16_t *input_channel_select;	    /**< For each FIR define in ch */
 	int16_t *output_channel_mix;	    /**< For each FIR define out ch */
 	int16_t *output_stream_mix;         /**< for each FIR define stream */
+	int count;
 	size_t fir_delay_size;              /**< allocated size */
 	bool config_ready;                  /**< set when fully received */
```

Comment:
wonder if maybe we should go the way of the linux kernel if you have a bunch of bools and switch to 1 width ints

Reply:
Oops, this an unused variable that I should remove. Did you mean that the bools would be better some other way?

Reply:
See here 1 width ints are used in the kernel rather than literal bools (i believe because bools use a whole byte anyways) where as these can be packed? (its been a while since i read the explanation so don't quote me on it.)
https://github.com/thesofproject/linux/blob/topic/sof-dev/include/sound/soc.h#L809

Reply:
Thanks, I'll try! If it's not increasing (much) code size it's OK for this component. None of the booleans are used in busy loops.